File: src/AzureDataFactory.TestingFramework/Models/Activities/ControlActivity/ExecutePipelineActivity.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

using AzureDataFactory.TestingFramework.Models.Base;
using AzureDataFactory.TestingFramework.Models.Pipelines;

namespace AzureDataFactory.TestingFramework.Models;

public partial class ExecutePipelineActivity : IIterationActivity
{
    internal List<IRunParameter> GetChildRunParameters(PipelineRunState state)
    {
        var parameters = state.Parameters.Where(p => p.Type == ParameterType.Global).ToList();
        parameters.AddRange(Parameters.Select(p => new RunParameter<string>(ParameterType.Pipeline, p.Key, p.Value)));
        return parameters;
    }
}

File: src/python/data_factory_testing_framework/models/activities/control_activities/execute_pipeline_activity.py
from typing import List

from data_factory_testing_framework.models.base.run_parameter import RunParameter
from data_factory_testing_framework.models.base.run_parameter_type import RunParameterType
from data_factory_testing_framework.models.state.pipeline_run_state import PipelineRunState


class ExecutePipelineActivity:
    @staticmethod
    def get_child_run_parameters(state: PipelineRunState) -> List[RunParameter]:
        child_parameters = []
        for parameter in state.parameters:
            if parameter.type == RunParameterType.Global:
                child_parameters.append(RunParameter(RunParameterType.Global, parameter.name, parameter.value))

        return child_parameters
